transport_nantes/mobilito/views.py

```diff
@@ -38,6 +38,132 @@ class TutorialView(TemplateView):
     """
 
     template_name = 'mobilito/tutorial.html'
+    # This dict is used to map tutorial pages to their next page.
+    all_tutorial_pages = {
+        'presentation': {
+            "last_modified": datetime(
+                year=2022, month=8, day=3, tzinfo=timezone.utc)
+        },
+        'pietons': {
+            "last_modified": datetime(
+                year=2022, month=8, day=3, tzinfo=timezone.utc)
+        },
+        'voitures': {
+            "last_modified": datetime(
+                year=2022, month=8, day=3, tzinfo=timezone.utc)
+        },
+        'velos': {
+            "last_modified": datetime(
+                year=2022, month=8, day=3, tzinfo=timezone.utc)
+        },
+        'transports-collectifs': {
+            "last_modified": datetime(
+                year=2022, month=8, day=3, tzinfo=timezone.utc)
+        },
+    }
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        tutorial_page = kwargs.get('tutorial_page', 'presentation')
+        if tutorial_page not in self.all_tutorial_pages:
+            # Trying to access a tutorial page that doesn't exist.
+            raise Http404()
+        context['active_tab'] = tutorial_page
+        self.template_name = f'mobilito/tutorial_{tutorial_page}.html'
+        # We're to use this pattern to make str <-> datetime conversions
+        self.strftime_pattern = "%Y-%m-%d %H:%M:%S"
```

Comment:
When you want an RFC 3399 or an ISO 8601 date, you should use the specific functions that give you that rather than specifying formats.

At least use "%F %T" if no other function is available.

Reply:
This pattern works for its purpose, I needed one that can make a string go back to datetime object and vice versa, it's not down to the ms sure, but it doesn't need to be, afaik.

What the change to make here ?

Reply:
The change (although, cf my code suggestion that avoids all this complexity) is just to say "%F %T" instead of "%Y-%m-%d %H:%M:%S".